include/mull/Filters/FilePathFilter.h

```diff
@@ -4,32 +4,32 @@
 #include "mull/Filters/InstructionFilter.h"
 #include "mull/Filters/MutationFilter.h"
 
+#include <llvm/Support/Regex.h>
 #include <mutex>
-#include <regex>
+#include <string>
+#include <utility>
 #include <unordered_map>
 #include <vector>
 
 namespace mull {
 struct SourceLocation;
 
-class FilePathFilter : public MutationFilter,
-                       public FunctionFilter,
-                       public InstructionFilter {
+class FilePathFilter : public MutationFilter, public FunctionFilter, public InstructionFilter {
 public:
   bool shouldSkip(MutationPoint *point) override;
   bool shouldSkip(llvm::Function *function) override;
   bool shouldSkip(llvm::Instruction *instruction) const override;
   bool shouldSkip(const std::string &sourceFilePath) const;
 
   std::string name() override;
-  void exclude(const std::string &filter);
-  void include(const std::string &filter);
+  std::pair<bool, std::string> exclude(const std::string &filter);
+  std::pair<bool, std::string> include(const std::string &filter);
 
 private:
   bool shouldSkip(const mull::SourceLocation &location) const;
 
-  std::vector<std::regex> includeFilters;
-  std::vector<std::regex> excludeFilters;
+  mutable std::vector<llvm::Regex> includeFilters;
```

Comment:
Why does it have to be mutable?

Reply:
Yes, it does. llvm::Regex does not offer const qualified member functions.

Reply:
Do you know from which versions it provides the const qualified functions? If it's only some "old" versions, can we wrap it in `#if/#endif` and drop later when we drop those versions support?

Reply:
If you prefers it that way I can have a look

Reply:
That would be really nice! I had some hard time debugging multithreading issues with `mutable` objects hidden down the line, so I'd prefer to avoid them whenever possible 😄

Reply:
the mutable seems to only be necessary as the interface itself is defined as const.